addons/gopaddle-lite/enable

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+KUBECTL="$SNAP/kubectl --kubeconfig=${SNAP_DATA}/credentials/client.config"
+
+HELM="$SNAP_DATA/bin/helm3 --kubeconfig=$SNAP_DATA/credentials/client.config"
+
+KUBE_VERSION=$($KUBECTL version | grep Server| cut -d '{' -f 2| cut -d ',' -f 2| cut -d ':' -f 2 |cut -d '+' -f 1|cut -d '"' -f 2)
+
+CURRENT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
+
+STATIC_VERSION="24"
+if [ "$KUBE_VERSION" -gt $STATIC_VERSION ]
+then 
+   echo "$KUBE_VERSION is not supported"
+fi
+ 
+
+STATICIP=""
+HELM_VERSION="4.2.4"
+STORAGECLASS=""
+ARGS=$(getopt -a --options i:v:s: --longoptions "ip:,version:,storageclass:" -- "$@")
+eval set -- "$ARGS"
+
+while true; do
+  case "$1" in
+    -i|--ip)
+      STATICIP="$2"
+      shift 2;;
+    -v|--version)
+      HELM_VERSION="$2"
+      shift 2;;
+    --)
+      break;;
+  esac
+done
+
+PUBLICIP=true
+if [ $STATICIP ]
+then
+   echo "static IP of the microk8s cluster: $STATICIP"
+   echo ""
+   "$SNAP/microk8s-enable.wrapper" host-access:ip=$STATICIP
+elif $KUBECTL get node -o yaml | grep ExternalIP >/dev/null 2>&1
+then 
+     echo "Static IP input is not provided. Assuming External IP of the microk8s node for the gopaddle access endpoint."
+     echo ""
+else
+     echo "Static IP input is not provided. External IP is not set for the microk8s node. Assuming Internal IP of the microk8s node for the gopaddle access endpoint."
+     PUBLICIP=false
+     echo ""
+fi
+
+if [ -z "$STORAGECLASS" ]
+then
+  run_with_sudo $KUBECTL apply -f $CURRENT_DIR/storageclass-retain.yaml
+  STORAGECLASS="microk8s-hostpath-gp-retain"
+fi
+
+
+do_prerequisites() {
+  refresh_opt_in_config "authentication-token-webhook" "true" kubelet
+  restart_service kubelet
+  # enable helm3 
+  "$SNAP/microk8s-enable.wrapper" helm3
+
+  # enable storage
+  "$SNAP/microk8s-enable.wrapper" hostpath-storage
+
+  # enable dns 
+  "$SNAP/microk8s-enable.wrapper" dns
+
+  # enable metrics-server
+  "$SNAP/microk8s-enable.wrapper" metrics-server
+   
+   run_with_sudo snap install jq
```

Comment:
Do not install jq in the user's machine this will not work in strict confinement. You could bring jq with the addon or find another way to parse the output you want.

Reply:
I have removed the dependency on jq. Instead, I will be using plain awk commands.